### PR TITLE
Move vae on hpu

### PR DESCRIPTION
**Pull request by libinta**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/stable-diffusion/train_text_to_image_sdxl.py [hunk 1]
```diff
@@ -877,12 +888,13 @@ def preprocess_train(examples):
         new_fingerprint = Hasher.hash(args)
         new_fingerprint_for_vae = Hasher.hash("vae")
 
-        train_dataset = train_dataset.map(compute_embeddings_fn, batched=True, new_fingerprint=new_fingerprint)
+        train_dataset = train_dataset.map(compute_embeddings_fn, batched=True, 
+                                          new_fingerprint=new_fingerprint,load_from_cache_file=False)
         train_dataset = train_dataset.map(
```
Comment (libinta) [inline examples/stable-diffusion/train_text_to_image_sdxl.py:893]: remove load_from_cache_file=False

File: examples/stable-diffusion/run_1x_bs16.sh [hunk 2]
```diff
@@ -17,10 +17,8 @@ python train_text_to_image_sdxl.py \
   --throughput_warmup_steps 3 \
   --dataloader_num_workers 8 \
   --bf16 \
-  --use_hpu_graphs_for_training \
```
Comment (libinta) [inline examples/stable-diffusion/run_1x_bs16.sh:20]: add this back

File: examples/stable-diffusion/train_text_to_image_sdxl.py [hunk 3]
```diff
@@ -568,22 +568,18 @@ def encode_prompt(batch, text_encoders, tokenizers, proportion_empty_prompts, ca
...
-def compute_vae_encodings(batch, vae):
-    images = batch.pop("pixel_values")
-    pixel_values = torch.stack(list(images))
+def compute_vae_encodings(pixel_values, vae):
     pixel_values = pixel_values.to(memory_format=torch.contiguous_format).float()
     pixel_values = pixel_values.to(vae.device, dtype=vae.dtype)
-
     with torch.no_grad():
         model_input = vae.encode(pixel_values).latent_dist.sample()
-    model_input = model_input * vae.config.scaling_factor
```
Comment (ssarkar2) [inline examples/stable-diffusion/train_text_to_image_sdxl.py:585]: this line is deleted?

**Review by ssarkar2 — APPROVED** [review]

File: examples/stable-diffusion/train_text_to_image_sdxl.py [hunk 4]
```diff
@@ -1219,17 +1220,16 @@ def compute_time_ids(original_size, crops_coords_top_left):
                         accelerator.save_state(save_path)
                         logger.info(f"Saved state to {save_path}")
 
-            if global_step % args.logging_step == 0:
-                train_loss_scalar = train_loss.item()
-                accelerator.log({"train_loss": train_loss_scalar}, step=global_step)
+                if (global_step - 1) % args.logging_step == 0 or global_step == args.max_train_steps:
```
Comment (yeonsily) [inline examples/stable-diffusion/train_text_to_image_sdxl.py:1223]: From the original https://github.com/huggingface/diffusers/blob/v0.25.1/examples/text_to_image/train_text_to_image_sdxl.py#L1141, logging is outside of accelerator.sync_gradients. Do wee need to put inside of sync_gradient?

**Review by yeonsily — APPROVED** [review]

**Pull request merged**